### DISCUSSION TITLE
when caps word is enabled it's impossible to type `-`

### DISCUSSION THREAD

**Discussion by idavydov** [manna-harbour/miryoku#202] [Q&A] [answered]
Hi,
I think caps word is a very useful feature.

I found a problem with it which is somewhat specific to miryoku. In this layout, it takes the same amount of effort to type `-` as `_`. However, caps word "simplifies" typing programming constants by replacing `-` with `_`. This makes it impossible to type `-`, when caps word is enabled.

I often encounter constants like `ABC-DEF`, and it makes it difficult to use caps word with them.

Is there maybe a way to change this behaviour? At least for QMK.

**Comment by manna-harbour** [comment, marked as answer]
Miryoku QMK and Miryoku ZMK use the default Caps Word configuration for the respective firmware. Both can be customised.

For QMK, the docs are https://github.com/qmk/qmk_firmware/blob/master/docs/feature_caps_word.md.

To prevent `-` from being shifted, you'd copy the definition of `caps_word_press_user` from https://github.com/qmk/qmk_firmware/blob/master/docs/feature_caps_word.md#configure-which-keys-are-word-breaking-idconfigure-which-keys-are-word-breaking, add it to https://github.com/manna-harbour/miryoku_qmk/blob/miryoku/users/manna-harbour_miryoku/manna-harbour_miryoku.c, and move the line containing `KC_MINS` from the first section to the second section. E.g. https://github.com/qmk/qmk_firmware/issues/18813.

Alternatively, use Caps Lock instead in this situation (Shift + Caps Word).

**Reply by idavydov** [reply]
Thanks a lot. I'm wondering whether this could be a default. Unless you do not want to override `caps_word_press_user` of course, for easier customization by users.

**Reply by manna-harbour** [reply]
It should probably be a built-in option at least, and perhaps should be the default.

**Comment by idavydov** [comment]
ok, I put this into `kaymap.c`, and it seems to help:

```
bool caps_word_press_user(uint16_t keycode) {
    switch (keycode) {
        // Keycodes that continue Caps Word, with shift applied.
        case KC_A ... KC_Z:
            add_weak_mods(MOD_BIT(KC_LSFT));  // Apply shift to next key.
            return true;

        // Keycodes that continue Caps Word, without shifting.
        case KC_1 ... KC_0:
        case KC_BSPC:
        case KC_DEL:
        case KC_UNDS:
        case KC_MINS:
            return true;

        default:
            return false;  // Deactivate Caps Word.
    }
}
```

But maybe this could be a default for miryoku?

**Reply by manna-harbour** [reply]
I think it'll depend on your use case, but as you say, it's still easy to type `_`, so perhaps it should be a built-in compile time option at least, if not the default.
